build.sh

```diff
@@ -121,14 +121,32 @@ else
   $PYTHON_EXECUTABLE -m pip install \
       --target=$ROOT_DIR/python/ray/pyarrow_files pyarrow==0.12.0.RAY \
       --find-links https://s3-us-west-2.amazonaws.com/arrow-wheels/9357dc130789ee42f8181d8724bee1d5d1509060/index.html
-  bazel build //:ray_pkg -c opt --verbose_failures
-  # Copy files and keep them writeable. This is a workaround, as Bazel
-  # marks all generated files non-writeable. If we would just copy them
-  # over without adding write permission, the copy would fail the next time.
-  # TODO(pcm): It would be great to have a solution here that does not
-  # require us to copy the files.
-  find $ROOT_DIR/bazel-genfiles/ray_pkg/ -exec chmod +w {} \;
-  cp -r $ROOT_DIR/bazel-genfiles/ray_pkg/ray $ROOT_DIR/python || true
+
+  if [ "$RAY_BUILD_JAVA" == "YES" ]; then
+    bazel build //:ray_java_pkg -c opt
+    #soft link
+    mkdir -p $ROOT_DIR/build/src/ray/raylet/
+    mkdir -p $ROOT_DIR/build/src/ray/gcs/redis_module/
+    mkdir -p $ROOT_DIR/build/src/ray/thirdparty/redis/src/
+    mkdir -p $ROOT_DIR/build/src/plasma/
+    ln -sf $ROOT_DIR/bazel-bin/* $ROOT_DIR/build/src/ray/raylet/
+    ln -sf $ROOT_DIR/bazel-bin/external/plasma/* $ROOT_DIR/build/src/plasma/
+    ln -sf $ROOT_DIR/bazel-genfiles/redis-server $ROOT_DIR/build/src/ray/thirdparty/redis/src/redis-server
+    ln -sf $ROOT_DIR/bazel-bin/ray_redis_module.so $ROOT_DIR/build/src/ray/gcs/redis_module/libray_redis_module.so
+    ln -sf $ROOT_DIR/bazel-bin/external/plasma/plasma_store_server $ROOT_DIR/build/src/plasma/plasma_store_server
+    ln -sf $ROOT_DIR/bazel-bin/raylet $ROOT_DIR/build/src/ray/raylet/raylet
+  fi
+
+  if [ "$RAY_BUILD_PYTHON" == "YES" ]; then
+    bazel build //:ray_pkg -c opt
```

Comment:
make sure to put all the arguments here (it has changed since this PR was originally created)